### PR TITLE
Upgrade sbt to 1.2.0

### PR DESCRIPTION
**Pull request by tanishiking**
This PR started to upgrade sbt to 1.2.0

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -228,7 +228,7 @@ def CiCommand(name: String)(commands: List[String]): Command =
       case (state, command) => ci(command) :: state
     }
   }
-def ci(command: String) = s"plz ${sys.env("CI_SCALA_VERSION")} $command"
+def ci(command: String) = s"++ ${sys.env("CI_SCALA_VERSION")} $command"
```
Comment (tanishiking) [inline build.sbt]: sbt-doge's `plz <version> <command>` migrated into `++ <version>! <command>` once. https://github.com/sbt/sbt/pull/2613
As sbt 1.2.0, `plz` seems to be compatible with `++` https://github.com/sbt/sbt/issues/3698

File: project/RunSbtCommand.scala [hunk 2]
```diff
@@ -1,28 +0,0 @@
-import sbt._
```
Comment (tanishiking) [inline project/RunSbtCommand.scala]: As far as I know, this code is not used now.

File: project/Dependencies.scala [hunk 3]
```diff
@@ -9,12 +9,11 @@ object Dependencies {
   val scalatestV = "3.2.0-SNAP10"
   val scalacheckV = "1.13.5"
   val coursier = "1.0.0-RC12"
-  val sbtV = "1.1.6"
```
Comment (tanishiking) [inline project/Dependencies.scala]: This line looks like being not refered.

File: project/RunSbtCommand.scala (hunk 2 above)
Comment (olafurpg) [reply project/RunSbtCommand.scala]: This is indeed no longer used after the sbt plugin was moved to a separate repo.

**Comment by tanishiking** [comment]
Although there seems no regressions in built document, it's ok to wait until the website is migrated to new one (it is really awsome:tada:). since this is not an urgent issue.

---

I took a diff of built documents between current master and this branch. (After running `$ sbt readme/run` in both environments)

```
$ diff -r -q scalameta/scalafmt/readme/target/scalatex/ tanishiking/scalafmt/readme/target/scalatex/
Files scalameta/scalafmt/readme/target/scalatex/index.html and tanishiking/scalafmt/readme/target/scalatex/index.html differ
Files scalameta/scalafmt/readme/target/scalatex/scripts.js and tanishiking/scalafmt/readme/target/scalatex/scripts.js differ
```

And it doesn't seem to have any regressions.

<details>
<summary>diff of index.html</summary>

`$ diff -u scalameta/scalafmt/readme/target/scalatex/index.html tanishiking/scalafmt/readme/target/scalatex/index.html`

```diff
--- scalameta/scalafmt/readme/target/scalatex/index.html	2018-08-02 14:34:28.000000000 +0900
+++ tanishiking/scalafmt/readme/target/scalatex/index.html	2018-08-02 14:30:28.000000000 +0900
@@ -62,7 +62,7 @@
 
   <link rel="stylesheet" href="custom.css" />
 
-<div style="margin: 0px;color: #333;text-align: center;padding: 2.5em 2em 0;border-bottom: 1px solid #eee;display: block;" id="Scalafmt" class=" scalatex-site-Styles-hoverContainer scalatex-site-Styles-headerTag"><h1 style="margin: 0.2em 0;font-size: 3em;font-weight: 300;">Scalafmt<a class=" scalatex-site-Styles-headerLink" href="#Scalafmt" style="position: absolute;right: 0px;"><i class="fa fa-link"></i></a></h1><br /><h2 style="font-weight: 300;color: #ccc;padding: 0px;margin-top: 0px;">0.6.6</h2></div><div class=" scalatex-site-Styles-content">
+<div style="margin: 0px;color: #333;text-align: center;padding: 2.5em 2em 0;border-bottom: 1px solid #eee;display: block;" id="Scalafmt" class=" scalatex-site-Styles-hoverContainer scalatex-site-Styles-headerTag"><h1 style="margin: 0.2em 0;font-size: 3em;font-weight: 300;">Scalafmt<a class=" scalatex-site-Styles-headerLink" href="#Scalafmt" style="position: absolute;right: 0px;"><i class="fa fa-link"></i></a></h1><br /><h2 style="font-weight: 300;color: #ccc;padding: 0px;margin-top: 0px;">1.5.1</h2></div><div class=" scalatex-site-Styles-content">
 
   
 <div>
@@ -150,17 +150,17 @@
 </p>
       <p>
         Create a standalone executable in <code>/usr/local/bin/scalafmt</code> with (sudo if necessary):
-        <pre><code class="xml scalatex-site-Styles-highlightMe">coursier bootstrap com.geirsson:scalafmt-cli_2.12:0.6.6 \
+        <pre><code class="xml scalatex-site-Styles-highlightMe">coursier bootstrap com.geirsson:scalafmt-cli_2.12:1.5.1 \
   -r bintray:scalameta/maven \
   -o /usr/local/bin/scalafmt --standalone --main org.scalafmt.cli.Cli
-scalafmt --version # should be 0.6.6
+scalafmt --version # should be 1.5.1
 </code></pre></p>
       <p>
         Alternatively you can create a slim 15 KiB bootstrap script with:
-        <pre><code class="xml scalatex-site-Styles-highlightMe">coursier bootstrap com.geirsson:scalafmt-cli_2.12:0.6.6 \
+        <pre><code class="xml scalatex-site-Styles-highlightMe">coursier bootstrap com.geirsson:scalafmt-cli_2.12:1.5.1 \
   -r bintray:scalameta/maven \
   -o scalafmt --main org.scalafmt.cli.Cli
-./scalafmt --version # should be 0.6.6
+./scalafmt --version # should be 1.5.1
 </code></pre>
         It is <b>recommended</b> to put this bootstrap script in your code repository
         to make sure everyone on your team, as well as CI, uses the same scalafmt
@@ -172,7 +172,7 @@
 </code></pre></p>
     <h2 id="Pre-release" style="display: block;" class=" scalatex-site-Styles-hoverContainer scalatex-site-Styles-headerTag"><a class=" scalatex-site-Styles-headerLink" href="#Pre-release" style="position: absolute;right: 0px;"><i class="fa fa-link"></i></a>Pre-release</h2>
       Our CI publishes a pre-release version of scalafmt to Bintray on every merge into master.
-      To use a pre-release, replace 0.6.6 with the version here:
+      To use a pre-release, replace 1.5.1 with the version here:
 
       
         <a href='https://bintray.com/scalameta/maven/scalafmt-cli/_latestVersion'><img src='https://api.bintray.com/packages/scalameta/maven/scalafmt-cli/images/download.svg'></a>
@@ -193,12 +193,12 @@
           Make sure you have a nailgun client installed. For example with <code>brew install nailgun</code>.</li>
         <li>
           Create a standalone executable in <code>/usr/local/bin/scalafmt_ng</code> with (sudo if necessary)
-          <pre><code class="xml scalatex-site-Styles-highlightMe">coursier bootstrap --standalone com.geirsson:scalafmt-cli_2.12:0.6.6 \
+          <pre><code class="xml scalatex-site-Styles-highlightMe">coursier bootstrap --standalone com.geirsson:scalafmt-cli_2.12:1.5.1 \
   -r bintray:scalameta/maven \
   -o /usr/local/bin/scalafmt_ng -f --main com.martiansoftware.nailgun.NGServer
 scalafmt_ng &amp; // start nailgun in background
 ng ng-alias scalafmt org.scalafmt.cli.Cli
-ng scalafmt --version # should be 0.6.6
+ng scalafmt --version # should be 1.5.1
 </code></pre></li></ul>
       <p>
         Nailgun keeps scalafmt running on a local server to avoid the JVM startup penalty
@@ -210,7 +210,7 @@
     <h2 id="Homebrew" style="display: block;" class=" scalatex-site-Styles-hoverContainer scalatex-site-Styles-headerTag"><a class=" scalatex-site-Styles-headerLink" href="#Homebrew" style="position: absolute;right: 0px;"><i class="fa fa-link"></i></a>Homebrew</h2>
       You can install scalafmt via Homebrew using a custom formula
         <pre><code class="scala scalatex-site-Styles-highlightMe">brew install --HEAD olafurpg/scalafmt/scalafmt
-scalafmt --version // should be 0.6.6
+scalafmt --version // should be 1.5.1
 
 // to upgrade between releases
 brew upgrade scalafmt
@@ -222,9 +222,9 @@
       may include flags that have not been published yet.
       <code class="scala scalatex-site-Styles-highlightMe" style="display: inline;padding: 0px;margin: 0px;"></code>
 
-      <pre><code class="scala scalatex-site-Styles-highlightMe">build commit: fd0df2a8d6ce94815fef87ff8257eb5f5e8d63e6
-build time: Thu Aug 02 14:33:43 JST 2018
-scalafmt 0.6.6-502-fd0df2a8
+      <pre><code class="scala scalatex-site-Styles-highlightMe">build commit: 4981e5a6129f2213b6350ab5110ba13d05526bbc
+build time: Thu Aug 02 14:28:55 JST 2018
+scalafmt 1.6.0-RC4-4-4981e5a6
 Usage: scalafmt [options] [&lt;file&gt;...]
 
   -h, --help               prints this usage text
@@ -323,7 +323,7 @@
     <h2 id="sbt-scalafmt" style="display: block;" class=" scalatex-site-Styles-hoverContainer scalatex-site-Styles-headerTag"><a class=" scalatex-site-Styles-headerLink" href="#sbt-scalafmt" style="position: absolute;right: 0px;"><i class="fa fa-link"></i></a>sbt-scalafmt</h2>
 
       <pre><code class="scala scalatex-site-Styles-highlightMe">// In project/plugins.sbt. Note, does not support sbt 0.13, only sbt 1.0.
-addSbtPlugin(&quot;com.geirsson&quot; % &quot;sbt-scalafmt&quot; % &quot;0.6.6&quot;)
+addSbtPlugin(&quot;com.geirsson&quot; % &quot;sbt-scalafmt&quot; % &quot;1.5.1&quot;)
 </code></pre>
       The sbt plugin is enabled by default for the <code>Test</code> and <code>Compile</code> configurations.
       To enable the plugin for integration tests
@@ -397,9 +397,9 @@
 
   <h1 id="Standalonelibrary" style="display: block;" class=" scalatex-site-Styles-hoverContainer scalatex-site-Styles-headerTag"><a class=" scalatex-site-Styles-headerLink" href="#Standalonelibrary" style="position: absolute;right: 0px;"><i class="fa fa-link"></i></a>Standalone library</h1>
     Add to your dependencies
-    <pre><code class="scala scalatex-site-Styles-highlightMe">libraryDependencies += &quot;com.geirsson&quot; %% &quot;scalafmt-core&quot; % &quot;0.6.6&quot;
+    <pre><code class="scala scalatex-site-Styles-highlightMe">libraryDependencies += &quot;com.geirsson&quot; %% &quot;scalafmt-core&quot; % &quot;1.5.1&quot;
 // Scala.js
-libraryDependencies += &quot;com.geirsson&quot; %%% &quot;scalafmt-core&quot; % &quot;0.6.6&quot;</code></pre>
+libraryDependencies += &quot;com.geirsson&quot; %%% &quot;scalafmt-core&quot; % &quot;1.5.1&quot;</code></pre>
     <p>
       Use the API like this
 </p>
@@ -1646,7 +1646,7 @@
       there are more configuration options if you visited nested fields
       like <code>spaces</code> and <code>newlines</code>.
 
-      <pre><code class="scala scalatex-site-Styles-highlightMe">version = &quot;0.6.6&quot;
+      <pre><code class="scala scalatex-site-Styles-highlightMe">version = &quot;1.5.1&quot;
 maxColumn = 80
 docstrings = scaladoc
 optIn.configStyleArguments = true
@@ -2609,7 +2609,7 @@
       be removed in 0.5.</li>
     <li>
       (Experimental) New module <code>scalafmt-bootstrap</code> that is an experiment to support
-      defining <code>version = 0.6.6</code> in <code>.scalafmt.conf</code>.
+      defining <code>version = 1.5.1</code> in <code>.scalafmt.conf</code>.
       The idea is to allow editor integrations (such as IntelliJ plugin)
       to dynamically use a specific scalafmt version for a specific project.
 
@@ -3586,4 +3586,4 @@
         );
       })
     })
-  </script></body></html>
\ No newline at end of file
+  </script></body></html>
\ No newline at end of file
```

</details>

**Comment by olafurpg** [comment]
It's great to hear that the website builds correctly. Before merging I also need to double-check that the intellij plugin still builds correctly. Since this is not an urgent priority, I propose to let this wait until the new website #1199 is merged.

File: build.sbt [hunk 4]
```diff
@@ -78,15 +78,15 @@ lazy val big = project
   .settings(
     allSettings,
     moduleName := "scalafmt-big",
-    crossScalaVersions := List(scala212),
+    ThisBuild / crossScalaVersions := List(scala212),
```
[the remark this answers is not in the document]
Comment (olafurpg) [reply build.sbt]: This is probably not necessary if we move `crossScalaVersions` into `inThisBuild(List(...))` at the top of the build.

File: build.sbt (hunk 1 above)
Comment (olafurpg) [reply build.sbt]: One of my favorite parts of sbt 1.0 is that sbt-doge is no longer needed 😄 (at least not for the basic use-case I had in the scalafmt build)

**Comment by olafurpg** [comment]
LGTM 👍 Thank you @tanishiking for upgrading, this ended up being helpful since the new website #1199 adds a dependency on sbt-tut, for which the latest version only supports sbt 1.

I will follow up with another PR to clean up the build further. The scalafmt build.sbt was the first sbt build I ever wrote! Now 3 years later I know better how to structure an sbt build :)

**Pull request merged**